File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) 
{
    vector<int> result;
    for(auto &v : commands)
    {
        vector<int> vec;
        for(int i=v[0] - 1; i < v[1]; i++)
        {
            vec.push_back(array[i]);
        }
        sort(vec.begin(), vec.end());
        result.push_back(vec[v[2] - 1]);
    }
    return result;
}

File: 프로그래머스/1/42748. K번째수/K번째수.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public int[] solution(int[] array, int[,] commands) 
    {
        List<int> temp = new List<int>();
        int[] result = new int[commands.GetLength(0)];
        for(int i=0; i<commands.GetLength(0); i++)
        {
            for(int j=commands[i,0]; j<=commands[i,1]; j++)
            {
                temp.Add(array[j - 1]);
            }
            int[] tmp = temp.ToArray();
            Array.Sort(tmp);
            result[i] = tmp[commands[i,2] - 1];
            temp.Clear();
        }
        return result;
    }
}